tests/rptest/tests/end_to_end.py

```diff
@@ -177,10 +177,12 @@ def has_finished_consuming():
                     return False
             return True
 
-        wait_until(has_finished_consuming,
-                   timeout_sec=timeout_sec,
-                   err_msg="Consumer failed to consume up to offsets %s after waiting %ds, last consumed offsets: %s." %\
-                   (str(last_acked_offsets), timeout_sec, list(self.last_consumed_offsets)))
+        wait_until(
+            has_finished_consuming,
+            timeout_sec=timeout_sec,
+            err_msg=lambda:
```

Comment:
Why is this a `lambda`?

Reply:
so that the string is interpolated at the time the test failed, this way it access an actual state, not the snapshot from the time the message being created.

Reply:
near